api/xds_protocol.rst

```diff
@@ -226,20 +226,61 @@ Transport API version
 In addition the resource type version described above, the xDS wire protocol has a
 transport version associated with it. This provides type versioning for messages such as
 :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` and :ref:`DiscoveryResponse
-<envoy_api_msg_DiscoveryResponse>`.
+<envoy_api_msg_DiscoveryResponse>`. It is also encoded in the gRPC method name, so a server
+can determine which version a client is speaking based on which method it calls.
 
-ACK/NACK and resource instance versioning
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Basic Protocol Overview
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Each xDS stream begins with a :ref:`DiscoveryRequest <envoy_api_msg_DiscoveryRequest>` from the
+client, which specifies the list of resources to subscribe to, the type URL corresponding to the
+subscribed resources, the node identifier, and an optional resource type instance version
+indicating the most recent version of the resource type that the client has already seen (see
+:ref:`ACK/NACK and resource type instance version <xds_ack_nack>` for details).
+
+The server will then send a :ref:`DiscoveryResponse <envoy_api_msg_DiscoveryResponse>` containing
+any resources that the client has subscribed to that have changed since the last resource type
+instance version that the client indicated it has seen. The server may send additional responses
+at any time when the subscribed resources change.
+
+Whenever the client receives a new response, it will send another request indicating whether or
+not the resources in the response were valid (see
+:ref:`ACK/NACK and resource type instance version <xds_ack_nack>` for details).
+
+Only the first request on a stream is guaranteed to carry the node identifier.
+The subsequent discovery requests on the same stream may carry an empty node
+identifier. This holds true regardless of the acceptance of the discovery
+responses on the same stream. The node identifier should always be identical if
+present more than once on the stream. It is sufficient to only check the first
+message for the node identifier as a result.
```

Comment:
fyi: this is broken for runtime type, found by accident. I think we may need to enforce it better in envoy before making the spec strict.

Reply:
This paragraph is actually not new; I just moved it from where it was before.  This is already part of the spec, so if something is breaking that requirement, I think it's unrelated to this PR.  (But I agree that this should be fixed.)